src/lib.rs

```diff
@@ -25,6 +25,9 @@
 //!     type DefinitionFuture = BoxFuture<Option<GotoDefinitionResponse>>;
 //!     type TypeDefinitionFuture = BoxFuture<Option<GotoDefinitionResponse>>;
 //!     type HighlightFuture = BoxFuture<Option<Vec<DocumentHighlight>>>;
+//!     type SignatureHelpFuture = BoxFuture<Option<SignatureHelp>>;
+//!     type GotoImplementationFuture = BoxFuture<Option<GotoImplementationResponse>>;
```

Comment:
This future should be renamed to `ImplementationFuture` to be consistent with `TypeDefinitionFuture` and should be moved to immediately below `TypeDefinitionFuture`.